### ISSUE TITLE
Basset doesn't work from View composers ?

### ISSUE THREAD

**Issue by Anahkiasen** [jasonlewis/basset#37]
It seems that anything registres with Basset withing a View Composer will be totally ignored, I don't know if that's a bug or if it's on purpose because of technical reasons ? I'm using view composers a lot to handle assets on a per-page basis, and I just thought this would seem like a logical place where Basset would be used.

**Comment by neoighodaro** [comment]
Yes i confirmed this too.

**Comment by jasonlewis** [comment]
Hm, I'm unable to reproduce this.

Are you running the latest version of Basset? I'm currently making a couple changes. Anything you can provide me with that might lead to why it's doing what it's doing?

**Comment by neoighodaro** [comment]
Alright, i already made away with the offending code and used Basset routes.php to set this instead. Perhaps i can explain. I wanted to have Basset routes set for specific layouts since i am using the Controller layout. So to reproduce this i think you should use a new installation of laravel 3.2.3 (latest version at the time) then create a Controller with a blade layout. Add a view composer for the layout and add the Basset route in that view composer. Try calling the set basset route from the layout.

**Comment by jasonlewis** [comment]
Thinking about it now... you'd need to declare an empty Basset route and then modify it in the composer.

Basset needs to the route to be created somewhere within the base of the application. Somewhere where it will always be triggered. So in either your own routes.php or start.php file, or within the Basset routes.php file.

This is because Basset actually creates a `Route::get()` on the fly which prevents any 404s from being thrown. This then allows a before filter to be set and the compiled assets to be rendered. It's only like this to allow Basset routes to be edited throughout an applications flow.

Basically you can keep your logic in the composer, but in another file (or the Basset routes.php file) you need to do the following.

``` php
<?php

Bundle::start('basset');

Basset::styles('example', function($basset){});
```

This registers the route so that it will always be available.

Hope that makes sense. I'm getting ready to push out another release which addresses a couple of other bugs as well.

**Comment by neoighodaro** [comment]
Yes it makes sense. I think just setting it in the Basset routes seems cleaner atm

**Comment by jasonlewis** [comment]
Okay cool, I'll wait to hear from @Anahkiasen first before closing this.
